code/datums/abilities/flock/flockmind.dm

```diff
@@ -167,12 +167,42 @@
 
 	logTheThing(LOG_COMBAT, holder.get_controlling_mob(), "designates [constructTarget(M)] as [flock.isEnemy(M) ? "" : "not "]an enemy at [log_loc(src.holder.owner)].")
 
+	flock.removeAlly(target)
+
 	if (flock.isEnemy(M))
 		flock.removeEnemy(M)
 		return
```

Comment:
```suggestion
	if (flock.isAlly(M))
		flock.removeAlly(M)
	else if (flock.isEnemy(M))
		flock.removeEnemy(M)
		return
```

Reply:
This would make it so casting designate ally on an enemy wouldn't mark them as ally